src/address/group.rs

```diff
@@ -0,0 +1,214 @@
+use std::hash::Hash;
+use std::ops::Deref;
+
+use crate::actor::Actor;
+use crate::address::envelope::ToEnvelope;
+use crate::address::{Addr, SendError, WeakAddr};
+use crate::handler::{Handler, Message};
+
+use weak_table::{traits::WeakElement, PtrWeakHashSet};
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+struct WeakTableAddrWrap<A: Actor>(Addr<A>);
+
+impl<A: Actor> Deref for WeakTableAddrWrap<A> {
+    type Target = Addr<A>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+#[derive(Debug)]
+struct WeakTableWeakAddrWrap<A: Actor>(WeakAddr<A>);
+
+impl<A: Actor> Deref for WeakTableWeakAddrWrap<A> {
+    type Target = WeakAddr<A>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<A: Actor> WeakElement for WeakTableWeakAddrWrap<A> {
+    type Strong = WeakTableAddrWrap<A>;
+
+    fn new(view: &Self::Strong) -> Self {
+        WeakTableWeakAddrWrap(view.0.downgrade())
+    }
+
+    fn view(&self) -> Option<Self::Strong> {
+        self.upgrade().map(|addr| WeakTableAddrWrap(addr))
+    }
+}
+
+impl<A: Actor> Clone for WeakTableWeakAddrWrap<A> {
+    fn clone(&self) -> WeakTableWeakAddrWrap<A> {
+        WeakTableWeakAddrWrap(self.0.clone())
+    }
+}
```

Comment:
Looks like this implementation can be derived. Less code to maintain is generally good.